tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+<?php
+
+
+use Drupal\Tests\UnitTestCase;
+use PHPUnit\Framework\TestCase;
+use Drupal\social_api\Utility\SocialApiImplementerInstaller;
+use Drupal\social_api\Annotation\Network;
+use Drupal\Drupal\social_api\User\UserAuthenticator;
+use Drupal\social_api\SocialApiDataHandler;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+// Controller
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\social_api\Plugin\NetworkManager;
+use Drupal\social_api\Controller\SocialApiController;
+use Drupal\Core\Plugin\DefaultPluginManager;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+//SocialApiException
+use Drupal\social_api\SocialApiException;
+// User
+use Drupal\social_api\User\UserManagerInterface;
+use Drupal\social_api\User\UserManager;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+//AuthManager
+use Drupal\social_api\AuthManager\OAuth2Manager;
+use Drupal\social_api\AuthManager\OAuth2ManagerInterface;
+//Settings
+use Drupal\social_api\Settings\SettingsBase;
+use Drupal\Core\Config\ImmutableConfig;
+use Drupal\Core\Config\Config;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Cache\Cache;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
+use Drupal\social_api\Settings\SettingsInterface;
+//Plugin
+use Drupal\social_api\Plugin\NetworkInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\social_api\Plugin\NetworkBase;
+
+
+
+class SocialApiTest extends UnitTestCase {
+  /**
+   * Checks the library required by an implementer.
+   *
+   * @param string $machine_name
+   *   The module machine name.
+   *
+   * @param string $name
+   *   The module name.
+   * @param string $library
+   *   The library machine name.
+   * @param float $min_version
+   *   The min version required.
+   * @param float $max_version
+   *   The max version required.
+   *
+   * @return array
+   *   Requirements messages.
+   */
+
+  public $machine_name;
+  public $name;
+  public $library;
+  public $min_version;
+  public $max_version;
+  protected $config;
+  private $networkManager;
```

Comment:
Is it really necessary to declare mock objects as properties? You just need them locally in the methods, so no need for them to be properties of the class.

Reply:
No, it's not necessary to declare them as properties.